raft.go

```diff
@@ -489,10 +492,14 @@ func (r *Raft) startStopReplication() {
 				notifyCh:            make(chan struct{}, 1),
 				stepDown:            r.leaderState.stepDown,
 			}
+
 			r.leaderState.replState[server.ID] = s
 			r.goFunc(func() { r.replicate(s) })
 			asyncNotifyCh(s.triggerCh)
 			r.observe(PeerObservation{Peer: server, Removed: false})
+		} else if ok && s.peer.Address != server.Address {
+			r.logger.Info("updating peer", "peer", server.ID)
+			s.peer = server
```

Comment:
I'm wondering if you need to call `asyncNotifyCh(s.triggerCh)` like we are above?

Reply:
Perhaps? What does that do? is it idempotent?

Reply:
it tells raft to kick off replication to this server

https://github.com/hashicorp/raft/blob/ff338ca008a48ab472f7a9e748ad10db15c462c9/replication.go#L156-L158

Reply:
I see. We're only change the peer address here. So it depends on if the replication layer will pick up this change or not? If we do call `asyncNotifyCh`, is this idempotent? Specifically, is there any harm in calling it?